src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,25 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	const int dirOld = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+	{
+		const double dirNew = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dirNew >= 0.0) ? 1 : -1;
+	}
 	else if (!_velocity.empty())
 	{
-		if (secIdx > 0)
-		{
-			const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx - 1));
-			if (dir < 0.0)
-				prevVelocity *= -1.0;
-		}
-
 		// We have both cross section area and interstitial flow rate
 		// _curVelocity has already been set to the network flow rate in setFlowRates()
 		// the direction of the flow (i.e., sign of _curVelocity) is given by _velocity
-		const double dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
-		if (dir < 0.0)
-			_curVelocity *= -1.0;
+		const double dirNew = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dirNew >= 0.0) ? 1 : -1;
 	}
+	if (dirOld * _dir < 0.0)
+		_curVelocity *= -1.0;
 
-	return (prevVelocity * static_cast<double>(_curVelocity) < 0.0);
+	return (dirOld * _dir < 0.0);
```

Comment:
This is a comparison `int` vs `double`. The result `dirOld * _dir` will be converted to `double` for the comparison. We can simplify this by comparing against `0` instead of `0.0`.

Reply:
I did that and just because I wanted to play around a bit more, I also tried the following:
```
const int dirNew = static_cast<int>(getSectionDependentScalar(_velocity, secIdx));
```
but I got this error:
```
/home/jo/software/cadet/CADET/src/libcadet/model/parts/ConvectionDispersionOperator.cpp:193:83: error: invalid static_cast from type ‘const sfad::Fwd<double>’ to type ‘int’
  193 |   const int dirNew = static_cast<int>(getSectionDependentScalar(_velocity, secIdx));
```
Why is that?

Reply:
`getSectionDependentScalar()` returns an object of type `active`. This object only implements an explicit conversion to `double`. This is why you can convert this custom type to `double` but not to anything else.